src/main/java/uk/gov/pay/products/persistence/dao/PaymentDao.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.pay.products.persistence.dao;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import uk.gov.pay.products.model.Payment;
+import uk.gov.pay.products.persistence.entity.PaymentEntity;
+
+import javax.persistence.EntityManager;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class PaymentDao extends JpaDao<PaymentEntity> {
+
+    @Inject
+    PaymentDao(Provider<EntityManager> entityManager) {
+        super(entityManager, PaymentEntity.class);
+    }
+
+    public Optional<PaymentEntity> findByExternalId(String externalId) {
+        String query = "SELECT payment FROM PaymentEntity payment " +
+                "WHERE payment.externalId = :externalId";
+
+        return entityManager.get()
+                .createQuery(query, PaymentEntity.class)
+                .setParameter("externalId", externalId)
+                .getResultList().stream().findFirst();
+    }
+
+    public List<Payment> findByProductId(Integer productId){
```

Comment:
For maximum flexibility it would be good to alway return entities from the DAO so that JPA can do its magic should we need to. Just following best practices as well.